pars-fmt/src/simple.rs

```diff
@@ -136,12 +145,41 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
-    fn into_matcher(self) -> Result<FmtMatcher<'a>, FormatError> {
-        let tokens = self.tokens;
+    /// Check that fields in the struct match up with fields in the fmt string.
+    fn validate_fields(&self, fields: &[String]) -> Result<(), FormatError> {
+        let num_fields = self.tokens.iter().filter(|t| t.is_field()).count();
+
+        if fields.len() != num_fields {
+            return Err(FormatError::new(
+                self.source,
+                0..self.source.len(),
+                format!("expected {} fields, found {}", fields.len(), num_fields),
+            ));
+        }
+
+        for field_range in self.tokens.iter().filter_map(|t| match t {
+            Token::Separator(_) => None,
+            Token::Field(ref rng) => Some(rng),
+        }) {
+            let field = &self.source[field_range.clone()];
+            if fields.iter().find(|s| *s == field).is_none() {
```

Comment:
I think we want an equality test here?

Reply:
@cmyr This is the main thing I think I'd change.

Reply:
I'm not sure I follow. This is taking each field in the struct declaration, and seeing if it exists in the format string, and erroring if any are missing. I think this makes sense?